docs/source/examples/spot-jobs.rst

```diff
@@ -147,7 +148,8 @@ With the above changes, you are ready to launch a spot job with ``sky spot launc
     $ sky spot launch -n bert-qa bert_qa.yaml
 
 SkyPilot will launch and start monitoring the spot job. When a preemption happens, SkyPilot will automatically
-search for resources across regions and clouds to re-launch the job.
+search for resources across regions and clouds to re-launch the job. The environment variable :code:`$SKYPILOT_SPOT_RUN_ID`
+can be used to identify the same run as it is identical across multiple preemptions/re-launchs for the same spot job.
```

Comment:
nit:

> To identify the same job across multiple recoveries, use the environment variable ``$SKYPILOT_SPOT_RUN_ID`` in the task's `run` commands or directly in the program itself (e.g., access via ``os.environ`` and pass to Weights & Biases for tracking purposes in your training script). It is made available to the task whenever it is invoked. The spot run ID is kept identical  across all recoveries of the same spot job.

Reply:
A question as I wrote the above -- what's the reason to write the glue code e.g., L131, rather than directly using ``os.environ`` in the train script? Is it because the former is more visible?

Reply:
The `--run_name` is natively supported by the huggingface repo, so there is no need to modify the code by using that option.
The run_name can definitely written in the code directly by using `os.environ`. I think one advantage for passing it through the argument is that, when the user wants to run it on on-demand VM, she can easily modify this `run_name` to any other costant name in the YAML file without changing it in the training code.

Reply:
Fixed the doc. Thanks!

Reply:
> I think one advantage for passing it through the argument is that, when the user wants to run it on on-demand VM, she can easily modify this run_name to any other costant name in the YAML file without changing it in the training code.

This reminds of the following: maybe we should make the run ID available regardless of on-demand or spot jobs. This way, no need for users to change their yaml? Can defer to a TODO.

Reply:
Good point! I renamed the variable to `SKYPILOT_RUN_ID` so that the on-demand can use the same name in the future.
Added a TODO in the controller.py